### ISSUE TITLE
Exception when editing code.

### ISSUE THREAD

**Issue by Jezza** [intellij-rust/intellij-rust#612]
I wasn't even doing much of noteworthy.
I created a new file, as was manually moving some code around.
If you have any more questions, I'd be happy to answer if I can.

```
parentByStub must not be null
java.lang.IllegalStateException: parentByStub must not be null
	at org.rust.lang.core.psi.impl.RustStubbedElementImpl.getParent(RustStubbedElementImpl.kt:18)
	at com.intellij.psi.impl.PsiElementBase.isValid(PsiElementBase.java:254)
	at com.intellij.extapi.psi.StubBasedPsiElementBase.isValid(StubBasedPsiElementBase.java:278)
	at com.intellij.psi.impl.PsiElementBase.isValid(PsiElementBase.java:258)
	at com.intellij.psi.impl.PsiElementBase.isValid(PsiElementBase.java:258)
	at com.intellij.extapi.psi.StubBasedPsiElementBase.isValid(StubBasedPsiElementBase.java:278)
	at com.intellij.psi.impl.PsiElementBase.isValid(PsiElementBase.java:258)
	at com.intellij.psi.impl.PsiElementBase.isValid(PsiElementBase.java:258)
	at com.intellij.psi.impl.PsiElementBase.isValid(PsiElementBase.java:258)
	at com.intellij.psi.impl.PsiElementBase.isValid(PsiElementBase.java:258)
	at com.intellij.psi.impl.PsiElementBase.isValid(PsiElementBase.java:258)
	at com.intellij.psi.impl.PsiElementBase.isValid(PsiElementBase.java:258)
	at com.intellij.psi.impl.PsiElementBase.isValid(PsiElementBase.java:258)
	at com.intellij.psi.impl.PsiElementBase.isValid(PsiElementBase.java:258)
	at com.intellij.psi.impl.PsiElementBase.isValid(PsiElementBase.java:258)
	at com.intellij.psi.impl.smartPointers.SmartPointerManagerImpl.updatePointerTarget(SmartPointerManagerImpl.java:260)
	at com.intellij.psi.impl.smartPointers.SmartPointerManagerImpl.updatePointerTargetsAfterReparse(SmartPointerManagerImpl.java:251)
	at com.intellij.psi.impl.PsiDocumentManagerBase.finishCommitInWriteAction(PsiDocumentManagerBase.java:344)
	at com.intellij.psi.impl.PsiDocumentManagerImpl.finishCommitInWriteAction(PsiDocumentManagerImpl.java:143)
	at com.intellij.psi.impl.PsiDocumentManagerBase$3.run(PsiDocumentManagerBase.java:298)
	at com.intellij.openapi.application.impl.ApplicationImpl.runWriteAction(ApplicationImpl.java:1010)
	at com.intellij.psi.impl.PsiDocumentManagerBase.finishCommit(PsiDocumentManagerBase.java:295)
	at com.intellij.psi.impl.DocumentCommitThread$5.run(DocumentCommitThread.java:528)
	at com.intellij.openapi.application.impl.LaterInvocator$FlushQueue.runNextEvent(LaterInvocator.java:337)
	at com.intellij.openapi.application.impl.LaterInvocator$FlushQueue.run(LaterInvocator.java:321)
	at java.awt.event.InvocationEvent.dispatch(InvocationEvent.java:311)
	at java.awt.EventQueue.dispatchEventImpl(EventQueue.java:756)
	at java.awt.EventQueue.access$500(EventQueue.java:97)
	at java.awt.EventQueue$3.run(EventQueue.java:709)
	at java.awt.EventQueue$3.run(EventQueue.java:703)
	at java.security.AccessController.doPrivileged(Native Method)
	at java.security.ProtectionDomain$1.doIntersectionPrivilege(ProtectionDomain.java:75)
	at java.awt.EventQueue.dispatchEvent(EventQueue.java:726)
	at com.intellij.ide.IdeEventQueue.defaultDispatchEvent(IdeEventQueue.java:866)
	at com.intellij.ide.IdeEventQueue._dispatchEvent(IdeEventQueue.java:654)
	at com.intellij.ide.IdeEventQueue.dispatchEvent(IdeEventQueue.java:381)
	at java.awt.EventDispatchThread.pumpOneEventForFilters(EventDispatchThread.java:201)
	at java.awt.EventDispatchThread.pumpEventsForFilter(EventDispatchThread.java:116)
	at java.awt.EventDispatchThread.pumpEventsForHierarchy(EventDispatchThread.java:105)
	at java.awt.EventDispatchThread.pumpEvents(EventDispatchThread.java:101)
	at java.awt.EventDispatchThread.pumpEvents(EventDispatchThread.java:93)
	at java.awt.EventDispatchThread.run(EventDispatchThread.java:82)



java.lang.IllegalStateException: parentByStub must not be null
	at org.rust.lang.core.psi.impl.RustStubbedElementImpl.getParent(RustStubbedElementImpl.kt:18)
	at com.intellij.psi.impl.PsiElementBase.isValid(PsiElementBase.java:254)
	at com.intellij.extapi.psi.StubBasedPsiElementBase.isValid(StubBasedPsiElementBase.java:278)
	at com.intellij.psi.impl.PsiElementBase.isValid(PsiElementBase.java:258)
	at com.intellij.psi.impl.PsiElementBase.isValid(PsiElementBase.java:258)
	at com.intellij.extapi.psi.StubBasedPsiElementBase.isValid(StubBasedPsiElementBase.java:278)
	at com.intellij.psi.impl.PsiElementBase.isValid(PsiElementBase.java:258)
	at com.intellij.psi.impl.PsiElementBase.isValid(PsiElementBase.java:258)
	at com.intellij.psi.impl.PsiElementBase.isValid(PsiElementBase.java:258)
	at com.intellij.psi.impl.PsiElementBase.isValid(PsiElementBase.java:258)
	at com.intellij.psi.impl.PsiElementBase.isValid(PsiElementBase.java:258)
	at com.intellij.psi.impl.PsiElementBase.isValid(PsiElementBase.java:258)
	at com.intellij.psi.impl.PsiElementBase.isValid(PsiElementBase.java:258)
	at com.intellij.psi.impl.PsiElementBase.isValid(PsiElementBase.java:258)
	at com.intellij.psi.impl.PsiElementBase.isValid(PsiElementBase.java:258)
	at com.intellij.psi.impl.smartPointers.SmartPointerManagerImpl.updatePointerTarget(SmartPointerManagerImpl.java:260)
	at com.intellij.psi.impl.smartPointers.SmartPointerManagerImpl.updatePointerTargetsAfterReparse(SmartPointerManagerImpl.java:251)
	at com.intellij.psi.impl.PsiDocumentManagerBase.finishCommitInWriteAction(PsiDocumentManagerBase.java:344)
	at com.intellij.psi.impl.PsiDocumentManagerImpl.finishCommitInWriteAction(PsiDocumentManagerImpl.java:143)
	at com.intellij.psi.impl.PsiDocumentManagerBase$3.run(PsiDocumentManagerBase.java:298)
	at com.intellij.openapi.application.impl.ApplicationImpl.runWriteAction(ApplicationImpl.java:1010)
	at com.intellij.psi.impl.PsiDocumentManagerBase.finishCommit(PsiDocumentManagerBase.java:295)
	at com.intellij.psi.impl.DocumentCommitThread$5.run(DocumentCommitThread.java:528)
	at com.intellij.openapi.application.impl.LaterInvocator$FlushQueue.runNextEvent(LaterInvocator.java:337)
	at com.intellij.openapi.application.impl.LaterInvocator$FlushQueue.run(LaterInvocator.java:321)
	at java.awt.event.InvocationEvent.dispatch(InvocationEvent.java:311)
	at java.awt.EventQueue.dispatchEventImpl(EventQueue.java:756)
	at java.awt.EventQueue.access$500(EventQueue.java:97)
	at java.awt.EventQueue$3.run(EventQueue.java:709)
	at java.awt.EventQueue$3.run(EventQueue.java:703)
	at java.security.AccessController.doPrivileged(Native Method)
	at java.security.ProtectionDomain$1.doIntersectionPrivilege(ProtectionDomain.java:75)
	at java.awt.EventQueue.dispatchEvent(EventQueue.java:726)
	at com.intellij.ide.IdeEventQueue.defaultDispatchEvent(IdeEventQueue.java:866)
	at com.intellij.ide.IdeEventQueue._dispatchEvent(IdeEventQueue.java:654)
	at com.intellij.ide.IdeEventQueue.dispatchEvent(IdeEventQueue.java:381)
	at java.awt.EventDispatchThread.pumpOneEventForFilters(EventDispatchThread.java:201)
	at java.awt.EventDispatchThread.pumpEventsForFilter(EventDispatchThread.java:116)
	at java.awt.EventDispatchThread.pumpEventsForHierarchy(EventDispatchThread.java:105)
	at java.awt.EventDispatchThread.pumpEvents(EventDispatchThread.java:101)
	at java.awt.EventDispatchThread.pumpEvents(EventDispatchThread.java:93)
	at java.awt.EventDispatchThread.run(EventDispatchThread.java:82)
```

**Comment by zjhmale** [comment]
seems like duplicate with https://github.com/intellij-rust/intellij-rust/issues/606

**Comment by Jezza** [comment]
Quite possibly.
Looking at the log, I get a number of similar exceptions as well.
And there's a lot...
This is just the list:

```
9:16:21 AM IllegalStateException: parentByStub must not be null
9:16:22 AM Platform and Plugin Updates: IntelliJ IDEA is ready to update.
9:16:23 AM IllegalStateException: parentByStub must not be null
9:25:07 AM IllegalStateException: parentByStub must not be null
9:25:08 AM IllegalStateException: parentByStub must not be null
9:25:10 AM IllegalStateException: parentByStub must not be null
9:25:17 AM IllegalStateException: parentByStub must not be null
9:25:23 AM IllegalStateException: parentByStub must not be null
9:25:48 AM IllegalStateException: parentByStub must not be null
9:25:57 AM IllegalStateException: parentByStub must not be null
9:26:10 AM IllegalStateException: parentByStub must not be null
9:26:14 AM IllegalStateException: parentByStub must not be null
9:26:33 AM IllegalStateException: parentByStub must not be null
9:26:42 AM IllegalStateException: parentByStub must not be null
9:26:46 AM IllegalStateException: parentByStub must not be null
9:26:55 AM IllegalStateException: parentByStub must not be null
9:27:12 AM IllegalStateException: parentByStub must not be null
9:27:16 AM IllegalStateException: parentByStub must not be null
9:27:20 AM IllegalStateException: parentByStub must not be null
9:27:23 AM IllegalStateException: parentByStub must not be null
9:27:24 AM IllegalStateException: parentByStub must not be null
9:27:26 AM IllegalStateException: parentByStub must not be null
9:27:28 AM IllegalStateException: parentByStub must not be null
9:27:29 AM IllegalStateException: parentByStub must not be null
9:27:35 AM IllegalStateException: parentByStub must not be null
9:28:30 AM IllegalStateException: parentByStub must not be null
9:29:16 AM IllegalStateException: parentByStub must not be null
9:29:20 AM IllegalStateException: parentByStub must not be null
9:29:31 AM IllegalStateException: parentByStub must not be null
9:29:57 AM IllegalStateException: parentByStub must not be null
9:30:13 AM IllegalStateException: parentByStub must not be null
9:30:15 AM IllegalStateException: parentByStub must not be null
9:30:17 AM IllegalStateException: parentByStub must not be null
9:30:39 AM IllegalStateException: parentByStub must not be null
9:31:30 AM IllegalStateException: parentByStub must not be null
9:31:33 AM IllegalStateException: parentByStub must not be null
9:31:47 AM IllegalStateException: parentByStub must not be null
9:31:52 AM IllegalStateException: parentByStub must not be null
9:32:08 AM IllegalStateException: parentByStub must not be null
9:32:11 AM IllegalStateException: parentByStub must not be null
9:32:24 AM IllegalStateException: parentByStub must not be null
9:32:47 AM IllegalStateException: Unable to serialize FileEditorManager state: parentByStub must not be null
9:33:18 AM IllegalStateException: parentByStub must not be null
9:33:20 AM IllegalStateException: parentByStub must not be null
9:33:26 AM IllegalStateException: parentByStub must not be null
9:34:23 AM IllegalStateException: parentByStub must not be null
9:34:25 AM IllegalStateException: parentByStub must not be null
9:35:13 AM IllegalStateException: parentByStub must not be null
9:35:14 AM IllegalStateException: parentByStub must not be null
9:35:27 AM IllegalStateException: parentByStub must not be null
9:35:30 AM IllegalStateException: parentByStub must not be null
9:35:34 AM IllegalStateException: parentByStub must not be null
9:36:05 AM IllegalStateException: parentByStub must not be null
9:36:19 AM IllegalStateException: parentByStub must not be null
9:36:47 AM IllegalStateException: Unable to serialize FileEditorManager state: parentByStub must not be null
9:37:49 AM IllegalStateException: parentByStub must not be null
9:37:56 AM IllegalStateException: parentByStub must not be null
9:38:20 AM IllegalStateException: parentByStub must not be null
9:38:33 AM IllegalStateException: parentByStub must not be null
9:39:58 AM IllegalStateException: Unable to serialize FileEditorManager state: parentByStub must not be null
9:40:28 AM IllegalStateException: Unable to serialize FileEditorManager state: parentByStub must not be null
9:40:43 AM IllegalStateException: Unable to serialize FileEditorManager state: parentByStub must not be null
9:41:26 AM IllegalStateException: parentByStub must not be null
9:41:29 AM IllegalStateException: parentByStub must not be null
9:41:33 AM IllegalStateException: parentByStub must not be null
9:41:41 AM IllegalStateException: parentByStub must not be null
9:41:59 AM IllegalStateException: parentByStub must not be null
9:42:02 AM IllegalStateException: parentByStub must not be null
9:42:09 AM IllegalStateException: parentByStub must not be null
9:42:16 AM IllegalStateException: parentByStub must not be null
9:42:18 AM IllegalStateException: parentByStub must not be null
9:42:23 AM IllegalStateException: parentByStub must not be null
9:42:27 AM IllegalStateException: parentByStub must not be null
9:42:29 AM IllegalStateException: parentByStub must not be null
9:42:41 AM IllegalStateException: parentByStub must not be null
9:42:47 AM IllegalStateException: Unable to serialize FileEditorManager state: parentByStub must not be null
9:43:42 AM IllegalStateException: Unable to serialize FileEditorManager state: parentByStub must not be null
9:44:37 AM IllegalStateException: Unable to serialize FileEditorManager state: parentByStub must not be null
9:45:47 AM IllegalStateException: Unable to serialize FileEditorManager state: parentByStub must not be null
9:46:28 AM IllegalStateException: Unable to serialize FileEditorManager state: parentByStub must not be null
9:46:29 AM IllegalStateException: Unable to serialize FileEditorManager state: parentByStub must not be null
9:46:33 AM IllegalStateException: Unable to serialize FileEditorManager state: parentByStub must not be null
9:46:37 AM IllegalStateException: Unable to serialize FileEditorManager state: parentByStub must not be null
9:46:40 AM IllegalStateException: Unable to serialize FileEditorManager state: parentByStub must not be null
9:47:11 AM IllegalStateException: Unable to serialize FileEditorManager state: parentByStub must not be null
9:47:45 AM IllegalStateException: Unable to serialize FileEditorManager state: parentByStub must not be null
9:48:12 AM IllegalStateException: parentByStub must not be null
9:48:17 AM IllegalStateException: parentByStub must not be null
9:48:43 AM IllegalStateException: parentByStub must not be null
9:48:51 AM IllegalStateException: Unable to serialize FileEditorManager state: parentByStub must not be null
9:49:15 AM IllegalStateException: Unable to serialize FileEditorManager state: parentByStub must not be null
9:50:30 AM IllegalStateException: parentByStub must not be null
9:50:37 AM IllegalStateException: parentByStub must not be null
9:50:41 AM IllegalStateException: parentByStub must not be null
9:50:44 AM IllegalStateException: parentByStub must not be null
9:50:47 AM IllegalStateException: parentByStub must not be null
9:50:52 AM IllegalStateException: parentByStub must not be null
9:50:54 AM IllegalStateException: parentByStub must not be null
9:50:57 AM IllegalStateException: parentByStub must not be null
9:51:00 AM IllegalStateException: parentByStub must not be null
9:51:11 AM IllegalStateException: parentByStub must not be null
9:51:19 AM com.intellij.diagnostic.MessagePool$TooManyErrorsException: Too many IDE fatal errors. Monitoring stopped.
9:51:19 AM com.intellij.diagnostic.MessagePool$TooManyErrorsException: Too many IDE fatal errors. Monitoring stopped.
```

Not all of those are the same exception.

**Comment by matklad** [comment]
Can you share a snippet of code that caused this exception? And can you reproduce it with some deterministic sequence of actions?

**Comment by Jezza** [comment]
What I was doing was so menial I can't even remember it.
I've been refactoring code all morning, but doing so manually.
So, I'm assuming it was when I copied something.

The code in question wouldn't be much help, as there's a lot going on.

**Comment by Jezza** [comment]
Decided I'll grab all of the exceptions that are thrown.
There might be some duplicates, I didn't check.
But there's 21 exceptions there.

https://drive.google.com/file/d/0B-ogKxquQHbqSDdXMmRJcjJpdXc/view?usp=sharing

**Comment by zummenix** [comment]
I have the same issue. Just edit, comment/uncomment and move code around.
Also sometimes I see a message:
```
Unable to save plugin settings: The plugin org.rust.lang failed to save settings and has been disabled. Please restart IntelliJ IDEA
```
-
```
IntelliJ IDEA 2016.1.4
Build #IC-145.2070, built on August 2, 2016
JRE: 1.8.0_76-release-b198 x86_64
JVM: OpenJDK 64-Bit Server VM by JetBrains s.r.o
```
```
Plugin Version: 0.0.1.931-nightly
```
```
cargo 0.13.0-nightly (9f8c469 2016-08-18)
rustc 1.13.0-nightly (499484f56 2016-08-18)
binary: rustc
commit-hash: 499484f56d336f4628a6fa016626beb3ef21ba81
commit-date: 2016-08-18
host: x86_64-apple-darwin
release: 1.13.0-nightly
```

**Comment by matklad** [comment]
I think I have identified the source of the issue, but I am not 100% sure yet, and the fix will take some time. 

The alpha version of the plugin should not be affected, you can downgrade to it.

**Comment by matklad** [comment]
I failed to reproduce this issue exactly, but parentByStub was definitely broken (#613). It is now fixed, so I hope this issue is also resolved. The fix is included in the latest nightly.

I'll close this, but please reopen if you still see the issue after the update!

**Comment by zummenix** [comment]
I can confirm that the issue is fixed for me on `Version: 0.0.1.945-nightly`. Thanks!

**Comment by matklad** [comment]
>I can confirm that the issue is fixed for me on Version: 0.0.1.945-nightly. Thanks!

Haven't you by any chance upgraded the IDE to 2016.2 in the meantime? Turns out that the `parentByStub` bug is unrelated to this exception. 

It happens with IDEA 15 and 2016.1, 2016.2 is not affected. Hope to fix it soon.

**Comment by zummenix** [comment]
@matklad, no, I haven't.
